presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeLimitWithDistinct.java

```diff
@@ -53,6 +53,7 @@
                         child.getSource(),
                         parent.getCount(),
                         false,
+                        child.getSource().getOutputSymbols(),
```

Comment:
I think currently logic here is also broken. When `DistinctLimitNode` is put, then all output symbols from `child` are used in distinct operator (not only group-by keys). It doesn't fail because of `PruneUnreferencedOutputs` and intermittent projections before group by (the code is incorrect though).